app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala

```diff
@@ -18,60 +18,53 @@ package uk.gov.hmrc.helptosavefrontend.controllers
 
 import javax.inject.Singleton
 
-import cats.data.{EitherT, OptionT, ValidatedNel}
+import cats.data.{EitherT, ValidatedNel}
 import cats.instances.future._
 import cats.syntax.either._
 import com.google.inject.Inject
-import play.api.Logger
 import play.api.i18n.{I18nSupport, MessagesApi}
-import play.api.mvc.Action
-import uk.gov.hmrc.helptosavefrontend.auth.HtsCompositePageVisibilityPredicate.twoFactorURI
+import play.api.mvc.{Action, AnyContent}
+import play.api.{Configuration, Environment, Logger}
 import uk.gov.hmrc.helptosavefrontend.connectors.NSIConnector.{SubmissionFailure, SubmissionResult, SubmissionSuccess}
 import uk.gov.hmrc.helptosavefrontend.connectors._
+import uk.gov.hmrc.helptosavefrontend.models.HtsAuth.HtsAuthRule
 import uk.gov.hmrc.helptosavefrontend.models.{HTSSession, NSIUserInfo, UserInfo}
-import uk.gov.hmrc.helptosavefrontend.models.UserInfo
-import uk.gov.hmrc.helptosavefrontend.services.userinfo.UserInfoService
+import uk.gov.hmrc.helptosavefrontend.services.HelpToSaveService
 import uk.gov.hmrc.helptosavefrontend.util.Result
 import uk.gov.hmrc.helptosavefrontend.views
-import uk.gov.hmrc.play.frontend.auth.AuthContext
-import uk.gov.hmrc.play.frontend.auth.connectors.domain.Accounts
 import uk.gov.hmrc.play.http.HeaderCarrier
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class RegisterController @Inject()(val sessionCacheConnector: SessionCacheConnector,
-                                   val messagesApi: MessagesApi,
-                                   eligibilityConnector: EligibilityConnector,
-                                   citizenDetailsConnector: CitizenDetailsConnector,
-                                   nSAndIConnector: NSIConnector) extends HelpToSaveController with I18nSupport {
-
-  val userInfoService = new UserInfoService(authConnector, citizenDetailsConnector)
-
-  def declaration =
-    AuthorisedHtsUserAction { implicit authContext ⇒
-      implicit request ⇒
-        validateUser(authContext).fold(
+class RegisterController @Inject()(override val messagesApi: MessagesApi,
+                                   htsService: HelpToSaveService,
+                                   nSAndIConnector: NSIConnector,
+                                   val sessionCacheConnector: SessionCacheConnector)
+  extends HelpToSaveAuth with I18nSupport {
+
+  def declaration: Action[AnyContent] = authorisedForHtsWithEnrolments(HtsAuthRule) {
+    implicit request ⇒
+      implicit userUrlWithNino ⇒
+        validateUser(userUrlWithNino.path.get, userUrlWithNino.nino.get).fold(
```

Comment:
Or you could change the type signature of `HtsActionWithEnrolments` from
```
case class UserDetailsUrlWithNino(path: Option[String], nino: Option[String])
HtsActionWithEnrolments = Request[AnyContent] => UserDetailsUrlWithNino => Future[Result]
```
to
```
case class UserDetailsUrlWithNino(path: String, nino: String)
HtsActionWithEnrolments = Request[AnyContent] => EitherT[Future,String,UserDetailsUrlWithNino] => Future[Result]
```
Then you could write
```
 def declaration: Action[AnyContent] = authorisedForHtsWithEnrolments(HtsAuthRule) {
    implicit request ⇒
      implicit userUrlWithNino ⇒
        for{
         UserDetailsUrlWithNino(path, nino) <- userUrlWithNino
         result                             <- validateUser(path, nino)
        } yield result.fold(..., ...)
```